modules/src/test_utils.rs

```diff
@@ -143,6 +143,50 @@ impl HostFunctionsProvider for Crypto {
     fn sha256_digest(data: &[u8]) -> [u8; 32] {
         sp_io::hashing::sha2_256(data)
     }
+
+    fn sha2_256(message: &[u8]) -> [u8; 32] {
+        sp_io::hashing::sha2_256(message)
+    }
+
+    fn sha2_512(message: &[u8]) -> [u8; 64] {
+        use sha2::Digest;
+        let mut hasher = sha2::Sha512::new();
+        hasher.update(message);
+        let hash = hasher.finalize();
```

Comment:
can you try using `finalize_into` instead?
https://docs.rs/sha2/latest/sha2/trait.Digest.html#tymethod.finalize_into

Reply:
Using that method will involve an extra line of code of creating the `GenericArray` first before passing it to `finalize_into`, even then,  I'll still need to do the `copy_from_slice` at the end since dereferencing the `GenericArray` will produce `&[u8]` not `[u8; LEN]`.